opensearch-project/OpenSearch-Dashboards
# [MDS] Support Vega Visualizations

### Description

This PR adds Vega support for MDS. If `data_source.enabled: true` and a `data_source_id` field is provided in the Vega spec, then the Vega visualization will render with data from the data source (assuming the data source has the specified index).

### Issues Resolved
closes #5927 

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/73027756/2f36633b-075a-4e0b-accd-a141e4ae28a7



## Testing the changes
### Pre-requisites
1. Enable MDS (set`data_source.enabled: true` in `opensearch_dashboards.yml`)
2. Add data source (in this example, it is `vegaDataSource`)
3. Add the newly created data source name to the `data_source_name` field under the `url` body

### Testing
1. Go to Visualize -> Create visualization -> Vega
2. Copy-paste existing Vega visualization from the data source (or create a new one)
3. Add a field `data_source_id` to the vega spec under `url` and set the value to the data source id.
5. Vega should render, update when timefield is updated, and save.

### Check List
